### ISSUE TITLE
CentOS 7 timeout issue re-appears.

### ISSUE THREAD

**Issue by Paladin** [svanzoest-cookbooks/apache2#360]
Getting a consistent timeout check when running against a local vagrant CentOS7 box. It's dying on the httpd -t syntax check. I was successfully able to repeat the command timeout from the command line after logging in to the vagrant box. I clocked it as taking nearly 30 seconds to return from the command line. I fixed it by adding a global ServerName directive in the httpd.conf file:

ServerName 127.0.0.1

What's happening is that when doing a syntax check on a config, the system is trying to come up with a fully-qualified domain name for the server, so you're stuck waiting for a dns lookup that fails.

http://linux.101hacks.com/unix/httpd-apr-sockaddr-info-get-error/ pointed me in the right direction for a fix.

Lots of options: create a servername.conf file in conf-available; write it directly in httpd.conf; or maybe create a sites-available file template without a virtual host directive to set it up. (I think I'll probably do that last as a workaround for now, until I see if you find a better way to do it.)

**Comment by Paladin** [comment]
Never mind. Cranial flatulence. That last won't work because the first time the timeout happens is before any of the site config files get written. So it has to be done while writing the main config file(s).

**Comment by drpebcak** [comment]
Not sure if I completely agree with this being an issue. If you aren't using a real FQDN/something with DNS, you can also just put an entry in your hosts file.

Did you have a suggestion for how the cookbook could address this without doing hacky stuff?

**Comment by Paladin** [comment]
If "hacky stuff" was better defined I'd have a shot at answering the question, but here's a few ideas. You decide how many of them are too 'hacky' to be permitted.

a) Include a ServerName 127.0.0.1 directive in the global area of the file.
b) If the cookbook user adds a "server_name" attribute to the apache node, write it out in the global file.

There are varying degrees of "hack" in both of those, with the second having the purist's advantage of only being present if the user wants/needs it there. And if you seriously want to keep the httpd.conf file genetically pure, you could take either of those two ideas above and refocus the output file, maybe using servername.conf instead of the main config file, and park it in the optional extra conf files dir. (That was actually what I did to get around the delay when I was checking it from the command line.)

When I get the chance today I'm going to look for a way to force the creation of a servername.conf file before the cookbook calls httpd -t for the first time (it escaped me the first time, but just because I didn't see it, doesn't mean it doesn't exist; I've had attacks of 'code blindness' before) because that seems to me to be the only way I might be able to solve this myself this side of forking the cookbook for my own use.

And when I'm (re)building production servers I do have FQDN's around. But sometimes I want to spin up a box (inside my desktop or in the cloud) just to try out an idea and see if it's going to be feasible. It's for those ad hoc use cases where fixing this would be useful. I know it's an edge case, but it's still useful.

**Comment by drpebcak** [comment]
I was thinking about this some more, and I realized why we haven't seen this issue in our own tests (not recently, anyway) - https://github.com/svanzoest-cookbooks/apache2/blob/master/.kitchen.yml#L60

We use the fqdn cookbook to solve this when testing in vagrant/test-kitchen type environments.

**Comment by drpebcak** [comment]
@Paladin as far as 'hacky' goes, I just wouldn't want anything to be added to the cookbook that could potentially filter into a production environment - existing users shouldn't have to change their workflows to avoid whatever addition is made.

**Comment by Paladin** [comment]
Hmmm. I'll have a look and see if it'll do what I need for this.

**Comment by Paladin** [comment]
Hmm. If my understanding is correct (not always a certainty) the fqdn cookbook solves my issue, but I had to use it slightly differently than you did. You included it in the definitions for the test suite, I need to include it in the platform, because both the tests and the build fail without it. So I added it to the CentOS 7 platforms. I think I need it for the Ubuntu 15 image I'm working with, too, but that's from my cloud provider, not opscode, so it's too early to say if it's needed there, as well.

I may pull it out and  manually include the recipe, eventually, simply because it may not be needed on anything but these test boxes; for the moment that's what I'm building in C7 and U15, so I'll jump that bridge when it gets to me.
